loader/linux/icd_linux.c

```diff
@@ -61,11 +71,34 @@ void khrIcdOsVendorsEnumerate(void)
         // attempt to load all files in the directory
         for (dirEntry = readdir(dir); dirEntry; dirEntry = readdir(dir) )
         {
+#if defined(DT_REG)
```

Comment:
This change is a bit hard to follow because it breaks up a switch statement with an `#ifdef`.  Would it be possible to refactor this code slightly to increase readability / maintainability?  I think this could look something like: Initialize a flag to "false", set it to "true" in the `#ifdef` code when appropriate, then check the flag outside of the `#ifdefs`.